custom_components/waste_collection_schedule/waste_collection_schedule/source/waipa_nz.py

```diff
@@ -0,0 +1,147 @@
+import datetime
+from waste_collection_schedule import Collection
+
+import requests
+import json
+from datetime import datetime
+
+TITLE = "Waipa District Council"
+DESCRIPTION = "Source for Waipa District Council. Finds both general and glass recycling dates."
+URL = "https://www.waipadc.govt.nz/"
+TEST_CASES = {
+    "10 Queen Street": {"address": "10 Queen Street"} #Monday
+    ,"1 Acacia Avenue": {"address": "1 Acacia Avenue"}#Wednesday
+}
+
+
+class Source:
+    def __init__(self, address):
+        self._address = address
+
+    def fetch(self):
+        entries = []
+
+        #initiate a session
+        url = "https://enterprise.mapimage.net/IntraMaps22A/ApplicationEngine/Projects/?configId=6aa41407-1db8-44e1-8487-0b9a08965283&appType=MapBuilder&project=b5bc138e-edce-4b01-b159-ec44539ab455&datasetCode="
+
+        payload={}
+        headers = {
+          'sec-ch-ua': '"Google Chrome";v="105", "Not)A;Brand";v="8", "Chromium";v="105"',
```

Comment:
Please add only the essential headers. Please reuse headers instead of redefining it multiple times.